File: ChatSupport/src/main/java/com/vantagecircle/chatapp/interfacePref/SharedPrefM.java
package com.vantagecircle.chatapp.interfacePref;

import android.content.Context;
import android.content.SharedPreferences;

import com.vantagecircle.chatapp.utils.Config;

/**
 * Created by bapidas on 11/07/17.
 */

public class SharedPrefM {
    private Context mContext;
    private SharedPreferences mSharedPreferences;
    private SharedPreferences.Editor mEditor;

    public SharedPrefM(Context mContext) {
        this.mContext = mContext;
    }

    public void saveString(String key, String value) {
        mSharedPreferences = mContext.getSharedPreferences(Config.APP_PREFS,
                Context.MODE_PRIVATE);
        mEditor = mSharedPreferences.edit();
        mEditor.putString(key, value);
        mEditor.apply();
    }

    public String getString(String key) {
        mSharedPreferences = mContext.getSharedPreferences(Config.APP_PREFS,
                Context.MODE_PRIVATE);
        return mSharedPreferences.getString(key, null);
    }

    public void clear(){
        mSharedPreferences = mContext.getSharedPreferences(Config.APP_PREFS,
                Context.MODE_PRIVATE);
        mEditor = mSharedPreferences.edit();
        mEditor.clear();
        mEditor.apply();
    }
}

File: app/src/main/java/com/bapidas/chattingapp/data/pref/SharedPrefM.kt
package com.bapidas.chattingapp.data.pref

import android.content.Context
import android.content.SharedPreferences
import com.bapidas.chattingapp.utils.Constants

/**
 * Created by bapidas on 11/07/17.
 */
class SharedPrefM(private val mContext: Context) {
    private var mSharedPreferences: SharedPreferences? = null
    private var mEditor: SharedPreferences.Editor? = null

    fun saveString(key: String?, value: String?) {
        mSharedPreferences = mContext.getSharedPreferences(Constants.APP_PREFS,
                Context.MODE_PRIVATE)
        mEditor = mSharedPreferences?.edit()
        mEditor?.putString(key, value)
        mEditor?.apply()
    }

    fun getString(key: String?): String? {
        mSharedPreferences = mContext.getSharedPreferences(Constants.APP_PREFS,
                Context.MODE_PRIVATE)
        return mSharedPreferences?.getString(key, null)
    }

    fun clear() {
        mSharedPreferences = mContext.getSharedPreferences(Constants.APP_PREFS,
                Context.MODE_PRIVATE)
        mEditor = mSharedPreferences?.edit()
        mEditor?.clear()
        mEditor?.apply()
    }

}